### ISSUE TITLE
Quest 7629 is bugged...

### ISSUE THREAD

**Issue by The-Great-Sephiroth** [mangosArchives/databaseZero_Rel19#30]
The quest "Imp Delivery" is bugged. I am a lv60 undead lock and finished getting the stardust. I then returned to Gorzeeki who presented me with the quest "Imp Delivery", but when I hit "Accept" I am told that I do not meet the requirements for the quest and now I cannot talk to him, he just tries to sell me stuff.

**Comment by The-Great-Sephiroth** [comment]
Alright I used GM commands to complete four other quests on a hunch. Before I can do "Imp Delivery" I have to do three quests given by the warlock near the Gorzeeki. After that, I have to get him three arcanite bars. Then he offers me the quest and I can accept it. This is a bug because he should not offer me that quest until I am eligible.

**Comment by Sinoxan** [comment]
I've looked into this and there is one comment on wowhead having the same problem as you and someone gives them the response of, it probably means you haven't completed the material quests, so I'm not sure if this is a bug.  EDIT: See Below

**Comment by Sinoxan** [comment]
Ok, it's set to automatically give you 7631, however you need to have completed the material quest in order to get that, so when it tries to give you quest 7631 it says you don't meet the requirements for it, it's not referring to quest 7629.  You have now completed 7629 and he now doesn't have any quests to offer so he only sells things.  I guess this can be closed now as this isn't a bug.

**Comment by The-Great-Sephiroth** [comment]
Actually it is a bug. On the official servers if you do not meet the requirements for a quest it is not offered. In other words, once I complete the quest it should stop if I have not done the others. It should not offer me the quest and then tell me that I do not meet the requirements. It should then show up once I finish the other three quests.

**Comment by Sinoxan** [comment]
If that's the case then quest 7629 isn't bugged, it's the server that's bugged, so it could do with a new issue in the server repository to make it easier to keep track of things.

**Comment by The-Great-Sephiroth** [comment]
So it is a code thing that is allowing quests that cannot be accepted to be given to the player? If that is the case then I will make a ticket there. The whole thing should be a simple fix, such as checking all prerequisites and if they are not met, simply not offering the quest.

**Comment by billy1arm** [comment]
If I understand the issue correctly, then it sounds like a database quest
issue.

i.e. Currently.. Complete Quest A, Quest C becomes available

but it should be. 

Complete Quest A, Quest B1,B2,B2 should become available, once all 3
complete, then C becomes available

Or

Complete Quest A, Quest B1,B2,B2 are next in chain - once all 3 complete,
then C becomes available

?

**Comment by The-Great-Sephiroth** [comment]
You are correct, Billy. After turning in the quest I should have not received any information or been offered the final quest in the chain. Instead, the other NPC should have lite up with an exclamation mark over his head and offered me the three quests needed. Then once those were turned in, the final quest would unlock.

**Comment by Sinoxan** [comment]
Sorry, I read your first post wrong, I understand the problem now and I'm working on a fix. :)

**Comment by Sinoxan** [comment]
https://github.com/Sinoxan/database-1/commit/5b7411c9143fe312733b6b73177fda9ff994054d

That should fix it.  However, I have no way of testing whether it actually does so if someone would be kind enough to test it and report back it would be greatly appreciated.  If not, I get a new PC in March so I can test it then if I still remember

**Comment by The-Great-Sephiroth** [comment]
I cannot test it with my lock who already did it, but my fiance is playing one. When she gets there she can test it if you haven't already.

**Comment by Sinoxan** [comment]
It's not on any live server at the moment.

**Comment by danielsreichenbach** [comment]
This requires a bit of research, and thus will be scheduled for release 2.0.8, since it can not be resolved for 2.0.7.

**Comment by danielsreichenbach** [comment]
Moved to the next release.
